mat64/dense.go

```diff
@@ -512,7 +512,13 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 	m.mat.Stride = int(cols)
 	m.capRows = int(rows)
 	m.capCols = int(cols)
-	m.mat.Data = use(m.mat.Data, m.mat.Rows*m.mat.Cols)
+
+	size := rows * cols
+	if rows < 0 || cols < 0 || int(size) < 0 || size > maxInt {
+		return matrix.ErrShape
+	}
+
+	m.mat.Data = use(m.mat.Data, int(size))
 
 	for i := range m.mat.Data {
 		err = binary.Read(buf, defaultEndian, &m.mat.Data[i])
```

Comment:
Why not use `binary.LittleEndian` directly instead of `defaultEndian` if the documentation explicitly says `little-endian encoded`?